core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -3602,7 +3602,10 @@ boolean isNull( Object data, boolean emptyStringDiffersFromNull ) throws KettleV
         return true;
       }
 
-      if ( emptyStringDiffersFromNull ) {
+      if ( value.equals( StringUtils.EMPTY ) && !emptyStringDiffersFromNull ) {
+        return true;
+      }
+      if ( value.equals( StringUtils.EMPTY ) && emptyStringDiffersFromNull ) {
         return false;
       }
```

Comment:
I believe these two `if` expressions could be simplified into a direct return of the condition:
```suggestion
      return value.equals(StringUtils.EMPTY) && !emptyStringDiffersFromNull;
```